### PR TITLE
46 parking button

### PR DESCRIPTION
**Pull request by um33**
Added the functionality to the smart car so  now it can park itself by clicking the parking button from the screen.

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java [hunk 1]
```diff
@@ -49,40 +50,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                    setAngle(STEERING_ANGLE, "Setting angel down right");
-                    setSpeed(-MOVEMENT_SPEED, "Moving down right");
-                } else if (dir == joystickJhr.stick_downLeft()) {
-                    setAngle(-STEERING_ANGLE, "Setting angel down left");
-                    setSpeed(-MOVEMENT_SPEED, "Moving down left");
-                } else if (dir == 0) {
-                    setAngle(STRAIGHT_ANGLE, "Setting angle straight");
-                    setSpeed(IDLE_SPEED, "Stopping smartcar");
-                }
+                float speed = (joystickJhr.joyY() / 150) * 100;
```
Comment (sh4r10) [inline SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java:53]: A setAngle method call is missing. This will break the joystick functionality.  please undo these changes and move them to a separate issue.

File: SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java [hunk 2]
```diff
@@ -144,9 +117,9 @@ public void messageArrived(String topic, MqttMessage message) throws Exception {
                         final byte[] payload = message.getPayload();
                         final int[] colors = new int[IMAGE_WIDTH * IMAGE_HEIGHT];
                         for (int ci = 0; ci < colors.length; ++ci) {
-                            final byte r = payload[3 * ci];
-                            final byte g = payload[3 * ci + 1];
-                            final byte b = payload[3 * ci + 2];
+                            final int r = payload[3 * ci] & 0xFF;
```
Comment (sh4r10) [inline SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java:120]: Can you please explain this to me?

**Review by sh4r10 — CHANGES_REQUESTED** [review]

File: SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java (hunk 2 above)
Comment (um33) [reply SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java:120]: this is related to branch 67-implement-smartcar-frontcam, i merged this branch with my branch because i wanted to see how much space camera takes on screen.

File: SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java (hunk 1 above)
Comment (um33) [reply SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java]: thank u for your review. i have fixed that

**Review by sh4r10 — DISMISSED** [review]

**Comment by sh4r10** [comment]
The joystick is unable to move due to the missing method call `Joystickjhr.move(moveEvent)`,  tested by @adityak714

**Review by adityak714 — APPROVED** [review]
The changes are clear, and were explained to me by @um33, as I sat beside him and he demonstrated the P button working. Camera does not work in the branch here, but upon merging with master, the vehicle_config was behind master so the issue will be fixed thereon.
